src/Libraries/RevitNodes/Elements/Views/View.cs

```diff
@@ -160,21 +160,26 @@ public override string ToString()
         #region Filter
 
         /// <summary>
-        /// Add Filter to View
+        ///     Add Filter to View. Filter will be added even if View has a View Template applied, which normally would prevent user from adding
+        ///     Filters without first disabling or modifying the View Template.
         /// </summary>
-        /// <param name="view">View</param>
         /// <param name="parameterFilter">Parameter filter</param>
-        public void AddFilter(Revit.Filter.ParameterFilterElement parameterFilter)
+        /// <returns name="view">View</returns>
+        public Element AddFilter(Revit.Filter.ParameterFilterElement parameterFilter)
         {
-            RevitServices.Transactions.TransactionManager.Instance.EnsureInTransaction(Application.Document.Current.InternalDocument);
-            this.InternalView.AddFilter(parameterFilter.InternalElement.Id);
-            RevitServices.Transactions.TransactionManager.Instance.TransactionTaskDone();
+            if (!this.InternalView.IsFilterApplied(parameterFilter.InternalElement.Id))
```

Comment:
If I'm reading this correctly, it seems that if there is already a filter applied to this view, then the `AddFilter` method will do nothing. Correct?